### PR TITLE
fix: [ANDROAPP-3857] Remove coordinates custom view

### PR DESCRIPTION
**Pull request by Balcan**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://jira.dhis2.org/browse/ANDROAPP-3857)

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

**Review by mmmateos — APPROVED** [review]

File: app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinateViewModel.java
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinateViewModel.java:159]: I think nobody is catching this throw?

File: app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinatesViewBindings.kt
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinatesViewBindings.kt:71]: maybe you could use the style provider? you don't have to do anything. It's already prepare and it would be something like val color = styleprovider.get[TextPrimary]

File: app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt:276]: why not having it in the viewModel or a mapper and be able to test it?

File: app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialActivity.java
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialActivity.java:971]: should we use a specific interface? empty methods like these break interface segregation  @andresmr???

File: app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialRepositoryImpl.java
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialRepositoryImpl.java:573]: is this dhis logic able to use from SDK?

File: app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialActivity.java
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialActivity.java:1040]: This is also repeated in other activities should we have a controller or something?

File: app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinateViewModel.java
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinateViewModel.java:207]: If lat and lon are null should we show observableErrorMessage.set(coordinatesErrorMessage) ??? Does it make sense to move  if (lat != null && lon != null) inside LngLatValidatorKt.areLngLatCorrect ????

File: app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinateViewModel.java
Comment (Balcan) [reply app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinateViewModel.java:116]: It can't crash, the extension functons already take care of nulls.

File: app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinateViewModel.java
Comment (Balcan) [reply app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinateViewModel.java:159]: True, this was done this way to avoid errors when using the coordinate view outside the form, like in the event initial screen

Comment (Balcan) [reply app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinateViewModel.java:159]: I'll remove it :)

File: app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinatesViewBindings.kt
Comment (Balcan) [reply app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinatesViewBindings.kt:71]: That would be great. I'll wait until it is merged to update it

File: app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt
Comment (Balcan) [reply app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt:276]: I'll move it to a helper class then, so we can reuse it

File: app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialActivity.java
Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialActivity.java:971]: Note that this is not in the form. This an individual usage of the coordinate layout in the event initial screen. I would not change more of this activity until we move all of it to this "first section" of the event form. We can create a ticket for this as soon as possible.

File: app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinateViewModel.java
Comment (Balcan) [reply app/src/main/java/org/dhis2/data/forms/dataentry/fields/coordinate/CoordinateViewModel.java:207]: Nop, they can be both null for example if the user deletes them. An error only needs to be show if:
- Only one is provided
- Both are provided but does not pass the validation

File: app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/data/forms/dataentry/FormView.kt:314]: 👍🏻

File: form/src/main/java/org/dhis2/form/ui/FormViewModel.kt
Comment (JaimeToca) [inline form/src/main/java/org/dhis2/form/ui/FormViewModel.kt:50]: empty¿

File: form/src/main/java/org/dhis2/form/ui/FormViewModel.kt
Comment (JaimeToca) [inline form/src/main/java/org/dhis2/form/ui/FormViewModel.kt:71]: I think unit test is missing 👍🏻

File: form/src/main/java/org/dhis2/form/data/GeometryController.kt
Comment (JaimeToca) [inline form/src/main/java/org/dhis2/form/data/GeometryController.kt:13]: I think unit test is missing. Perhaps you could pass Gson as default parameter in the constructor? 

Do we want to serialize nulls? https://futurestud.io/tutorials/gson-builder-force-serialization-of-null-values

File: app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialRepositoryImpl.java
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialRepositoryImpl.java:572]: I think unit test is missing 👍🏻 . (this leads to the previous question for having an eassier test)

**Review by JaimeToca — CHANGES_REQUESTED** [review]
Thanks for all the work and changes @Balcan 👍🏻  I added a few comments (most of them are related to unit tests)

File: form/src/main/java/org/dhis2/form/ui/FormViewModel.kt
Comment (Balcan) [reply form/src/main/java/org/dhis2/form/ui/FormViewModel.kt:50]: Removed!

File: app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialRepositoryImpl.java
Comment (Balcan) [reply app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialRepositoryImpl.java:572]: added!

**Review by JaimeToca — APPROVED** [review]

**Comment by JaimeToca** [comment]
Awesome work !

**Review by andresmr — APPROVED** [review]

**Pull request merged**
